pkg/docker/config/config_test.go

```diff
@@ -61,40 +50,35 @@ func TestGetPathToAuth(t *testing.T) {
 		{nil, linux, tmpDir + "/thisdoesnotexist", "", false},
 		{nil, darwin, tmpDir + "/thisdoesnotexist", darwinDefault, false},
 	} {
-		if c.xrd != "" {
-			os.Setenv("XDG_RUNTIME_DIR", c.xrd)
-		} else {
-			os.Unsetenv("XDG_RUNTIME_DIR")
-		}
-		res, lf, err := getPathToAuthWithOS(c.sys, c.os)
-		if c.expected == "" {
-			assert.Error(t, err)
-		} else {
-			require.NoError(t, err)
-			assert.Equal(t, c.expected, res)
-			assert.Equal(t, c.legacyFormat, lf)
-		}
+		t.Run(fmt.Sprintf("%d", caseIndex), func(t *testing.T) {
+			t.Setenv("XDG_RUNTIME_DIR", c.xrd)
+			if c.xrd == "" {
```

Comment:
This seems a little weird, why not only set XDG_RUNTIME_DIR if c.xrd is set? Unless you are worried about XDG_RUNTIME_DIR already being set.

Reply:
Thanks for pointing this out, this really deserves a comment. Updated now, in both places.

(`t.Setenv`, unlike `os.Setenv`, ensures the environment variable is always restored to the original value (or original unset state) after the test finishes; and there is no `t.Unsetenv`. So we call `t.Setenv` unconditionally to get get the cleanup guarantee, and only then worry about possibly unsetting the variable if wanted by the test.)